.changeset/stupid-cheetahs-sell.md

```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/vercel': patch
+---
+
+Update supported Node versions and add `beta` option.
```

Comment:
It would be great to add a code block to show how users can opt-in this

Reply:
I included a section in the `README.md` of where users can look to change the Node.js version.

Reply:
Would it be possible to have a  code block too?

Reply:
I don't think so, as you can only pick a Node.js version inside the project page settings.

Reply:
I mean a code block to show how to use this new option, from the configuration file `astro.config.mjs`

Reply:
I made requests to update how things are worded, nothing can actually be changed from the config so no code block required.

Reply:
Node version is set via `engines` in `package.json`, nothing to configure in `astro.config.mjs`.

Reply:
I thought the `validateRuntime` would take the node version somehow on Vercel CIs, but it's always taking it from the package.json?

Reply:
This is very confusing on Vercel's part. Vercel project specifies one Node.js version for the build step. That's the one set in Vercel project settings UI and can be overridden by `engines` in `package.json`. But then each serverless function has to [specify its own runtime](https://vercel.com/docs/build-output-api/v3/primitives#serverless-function-configuration). Function runtime can specify a different version from the project build version, and also can be another language altogether, e.g. Ruby or GO. In Astro context, it only makes sense to use Node.js as a runtime and probably only the project build version, which is the current Node version running during build.

Another complication is that the user runs the integration locally and can have a version newer than the last supported Vercel version and that's what this code needs to validate.

Reply:
I think such discussions are better kept public and accessible on GitHub